src/accountsdb/index.zig

```diff
@@ -291,6 +299,151 @@ pub const AccountIndex = struct {
             .parent => |parent| parent.next_ptr = if (parent.next_ptr) |ref| ref.next_ptr else null,
         }
     }
+
+    pub fn loadFromDisk(self: *Self, dir: std.fs.Dir, logger: sig.trace.Logger) !void {
+        const scoped_logger = logger.withScope("load index state");
+
+        // make sure all the appropriate file exist
+        try dir.access("manager_memory", .{});
+        try dir.access("manager_records", .{});
+        try dir.access("pubkey_ref_map", .{});
```

Comment:
Also, possible design suggestion: how about just having one file containing the data for all three of these section, wherein each section starts with a little-endian `u64` representing the length of the respective section? Ie:
```
[0..8]                   = manager_memory_len
[0..manager_memory_len]  = manager_memory
[0..8]                   = manager_records_len
[0..manager_records_len] = manager_records
[0..8]                   = pubkey_ref_map_len
[0..pubkey_ref_map_len]  = pubkey_ref_map
```
This would allow loading open a single file, and do a single mmap:
```zig
const index_state_file = dir.openFile("index-state.bin", .{});
defer index_state_file.close();

const index_state_mem = std.posix.mmap(...);
defer std.posix.munmap(index_state_mem);

var end: usize = 0;
const manager_memory_mem, end = readSection(index_state_mem, end);
const manager_records_mem, end = readSection(index_state_mem, end);
const pubkey_ref_map_mem, end = readSection(index_state_mem, end);

// parse manager_memory, manager_records, and pubkey_ref_map
```

where:
```zig
fn readSection(memory: []u8, start: usize) struct { []u8, usize } {
    const section_len_buf = memory[start..][0..@sizeOf(u64)];
    const section_len = std.mem.readInt(u64, section_len_buf, .little);
    return .{
        memory[@sizeOf(u64)..][0..section_len],
        start + @sizeOf(u64) + section_len,
    };
}
```
Same sort of deal while saving to disk just in reverse.

Reply:
6ee69f1d